src/server/qgsserver.cpp

```diff
@@ -396,47 +396,64 @@ void QgsServer::handleRequest( QgsServerRequest &request, QgsServerResponse &res
 
     try
     {
-      // TODO: split parse input into plain parse and processing from specific services
-      requestHandler.parseInput();
-    }
-    catch ( QgsMapServiceException &e )
-    {
-      QgsMessageLog::logMessage( "Parse input exception: " + e.message(), QStringLiteral( "Server" ), Qgis::MessageLevel::Critical );
-      requestHandler.setServiceException( e );
-    }
+      if ( response.feedback() && response.feedback()->isCanceled() )
+      {
+        throw QgsServerException( QStringLiteral( "Remote socket has been closed!" ) );
```

Comment:
If the request is canceled, this is not an error and doesn't deserve an exception IMHO. We should let the workflow proceed without error

Reply:
done!